skbio/embedding/_embedding.py

```diff
@@ -86,15 +110,212 @@ def __repr__(self):
+
+        if isinstance(sequence, Sequence):
+            sequence = str(sequence)
+        if isinstance(sequence, str):
+            sequence = sequence.encode("ascii")
+
+        vector = np.atleast_2d(vector)
+        if vector.shape[0] != 1:
+            raise ValueError("Only 1 vector per sequence is allowed.")
+
+        seq = np.array([sequence], dtype='O')
+        super(SequenceVector, self).__init__(vector, seq, **kwargs)
+
+    @property
+    def sequence(self):
+        r""" String representation of the underlying sequence """
+        return str(self)
+
+    def __repr__(self):
+        r"""
+        Return a string representation of the SequenceEmbedding object.
+
+        Returns
+        -------
+        str
+            A string representation of the SequenceEmbedding object.
+
+        See Also
+        --------
+        Protein
+        """
+        seq = Sequence(str(self))
+        rstr = _repr_helper(
+            repr(seq), "Sequence", "SequenceVector", "vector",
+            regex_match="length", shape=self.embedding.shape[1]
         )
         return rstr
+
+
+def embedding_vectors_to_numpy(embedding_vectors, validate=True):
+    r""" Convert an iterable of EmbeddingVector objects to a numpy array.
+
+    Parameters
+    ----------
+    embedding_vectors : iterable of EmbeddingVector objects
+        An iterable of EmbeddingVector objects, or objects that
+        subclass EmbeddingVector.
+    validate : bool, optional
+        If True, validate that all vectors have the same length
+        and are valid types
+
+    Returns
+    -------
+    np.ndarray
+        A numpy array of shape (n_sequences, n_features) where
+
+    Raises
+    ------
+    ValueError
+        If the vectors do not have the same length.
+
+    Notes
+    -----
+
+    """
+    if validate:
+        subcls = [issubclass(type(ev), EmbeddingVector) for ev in embedding_vectors]
+        if not all(subcls):
+            raise ValueError("Input iterable contains objects that "
+                             "do not subclass EmbeddingVector.")
+
+        types = [type(ev) for ev in embedding_vectors]
+        if not all(t == types[0] for t in types):
+            raise ValueError("All objects must be of the same type.")
+
+        lens = [len(ev.vector) for ev in embedding_vectors]
+        if not all(ln == lens[0] for ln in lens):
+            raise ValueError("All vectors must have the same length.")
+    data = np.vstack([ev.vector for ev in embedding_vectors])
+    return data
+
+
+def embedding_vectors_to_distance_matrix(embedding_vectors, metric='euclidean',
+                                         validate=True):
+    r""" Convert an iterable of EmbeddingVector objects to a
+    DistanceMatrix object.
+
+    Parameters
+    ----------
+    embedding_vectors : iterable of EmbeddingVector objects
+        An iterable of EmbeddingVector objects, or objects that
+        subclass EmbeddingVector.
+    metric : str, optional
```

Comment:
`metric` can be upgraded to support a user function. But it can be postponed to a future PR.